tests/zendesk/test_zendesk_source.py

```diff
@@ -166,6 +167,32 @@ def test_incrementing(destination_name: str) -> None:
     assert load_table_counts(pipeline_incremental, *INCREMENTAL_TABLES) == counts
 
 
+@pytest.mark.parametrize("destination_name", ALL_DESTINATIONS)
+def test_incremental_end_time(destination_name: str) -> None:
+    """Test chunk loading tickets with incremental end_value"""
+    pipeline = dlt.pipeline(
+        destination=destination_name,
+        full_refresh=True,
+    )
+
+    incremental_end_time = pendulum.DateTime(2023, 2, 7).in_tz("UTC")
+    data = zendesk_support(
+        incremental_start_time=pendulum.DateTime(2023, 2, 6).in_tz("UTC"),
+        incremental_end_time=incremental_end_time,
+    ).with_resources("tickets")
+
+    info = pipeline.run(data, write_disposition="replace")
+    assert_load_info(info)
+
+    with pipeline.sql_client() as client:
+        rows = [
+            pendulum.instance(row[0])
+            for row in client.execute_sql("SELECT updated_at FROM tickets")
+        ]
+
+    assert all(value < incremental_end_time for value in rows)
+
```

Comment:
what happens if we run the pipeline again, starting from `incremental_end_time`? will we get all records? could you test it?

another thing we should test is if incremental_end_time is exactly a timestamp of an ticket, then are we getting it twice? this is more like is the range open or closed. please add such information to docs after you figure it out (I think the answer is in the zendesk api)

Reply:
Added second run.  

The open/closed range is tested in dlt. So don't think it needs another test here, right? But I added the info to the docstring in the sources.

Reply:
we do not need another test, just:
1. set `start_time` to exact timestamp of first ticker in our test data and make sure that we have this ticket in our db
2. set `end_time` to exact timestamps of one of the tickets in the middle of our test data and make sure it is in our db exactly once

I know this is a little bit of PITA but we are testing the API behavior (and also double testing our incremental exclusiveness)

Reply:
Aha got you, good to have this. I'll add it!